core/store/orm/orm.go

```diff
@@ -419,6 +420,28 @@ func (orm *ORM) Sessions(offset, limit int) ([]models.Session, error) {
 	return sessions, err
 }
 
+// GetConfigValue returns the value for a named configuration entry
+func (orm *ORM) GetConfigValue(field string, value encoding.TextUnmarshaler) error {
+	name := EnvVarName(field)
+	config := models.Configuration{}
+	if err := orm.DB.First(&config, "name = ?", name).Error; err != nil {
+		return err
+	}
+	return value.UnmarshalText([]byte(config.Value))
+}
+
+// SetConfigValue returns the value for a named configuration entry
+func (orm *ORM) SetConfigValue(field string, value encoding.TextMarshaler) error {
+	name := EnvVarName(field)
+	textValue, err := value.MarshalText()
+	if err != nil {
+		return err
+	}
+	return orm.DB.Where(models.Configuration{Name: name}).
+		Assign(models.Configuration{Name: name, Value: string(textValue)}).
+		FirstOrCreate(&models.Configuration{}).Error
```

Comment:
nitpick: based on the example below, you would want the `Assign(models.Configuration...` to only have `Value`. As is right now, it might update `Name`, the functional PK.

http://gorm.io/docs/query.html#Assign-1

```golang
	return orm.DB.Where(models.Configuration{Name: name}). // <--- Where clause still matches on Name
		Assign(models.Configuration{Value: string(textValue)}). // <--- removed Name
		FirstOrCreate(&models.Configuration{}).Error
```

Based off of reading, haven't test it.

Reply:
`gorm` does not use the `Where` clause to populate the `Create`.